src/cursor.c

```diff
@@ -206,16 +206,14 @@ Cursor *Cursors_Reserve(CursorList *cl, StrongRef global_spec_ref, unsigned inte
   // If the cursor should be associated with a spec,
   // we assume that global_spec_ref points to a valid spec, else the function returns NULL.
   IndexSpec *spec = StrongRef_Get(global_spec_ref);
-
   // If we are in a coordinator ctx, the spec is NULL
-  if (spec && (spec->activeCursors >= spec->cursorsCap)) {
+  if (spec && spec->activeCursors >= RSGlobalConfig.indexCursorLimit) {
     /** Collect idle cursors now */
     Cursors_GCInternal(cl, 0);
-    if (spec->activeCursors >= spec->cursorsCap) {
-      QueryError_SetError(status, QUERY_ELIMIT, "Too many cursors allocated for index");
+    if (spec->activeCursors >= RSGlobalConfig.indexCursorLimit) {
+      QueryError_SetErrorFmt(status, QUERY_ELIMIT, "INDEX_CURSOR_LIMIT of %lld has been reached for an index", RSGlobalConfig.indexCursorLimit);
```

Comment:
I think the receiver of the error will know which index it is about since he issues the query request.
If the receiver of the error chooses to output the index name on any error it gets it can cause redundant information to be shown to the user.